### DISCUSSION TITLE
[Question]: Possibility to create an user using environment variables

### DISCUSSION THREAD

**Discussion by chnyda** [danny-avila/LibreChat#1886] [Q&A]
### What is your question?

I'm curious to find out if this application includes a feature that enables the automatic creation of an initial user account with a default password when it's launched for the first time.

### More Details

Is there a built-in mechanism that facilitates the setup of a default user, including setting up a password, to streamline the initial setup process and allow immediate access to the application? This would be particularly helpful for initial deployments or testing environments.

### What is the main subject of your question?

Other

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
If you have npm installed (local setup) and have the MONGO_URI setup, you can run `npm run create-user`

with docker, you have to enter the container shell and run the same command

**Reply by heurainbow** [reply]
not working anymore after recent update when I enter LibreChat-API container

**Reply by gitGunnl** [reply]
npm run create-user is not working anymore

**Reply by danny-avila** [reply]
I just tried it, it's working. This is assuming you are using the `deploy-compose.yml` file:

1. Exec into the container:
   ```
   docker exec -it LibreChat-API /bin/sh
   ```

2. Navigate up one directory:
   ```
   cd ..
   ```

3. Run the create-user script:
   ```
   npm run create-user
   ```

To combine these steps into a single command that you can run from your host machine:

```
docker exec -it LibreChat-API /bin/sh -c "cd .. && npm run create-user"
```

This command will exec into the container, change to the parent directory, and run the create-user script in one go.

**Reply by light299792458** [reply]
I see, i did not do it right. Thank you.

**Comment by danny-avila** [comment]
Closing since I've updated the documentation to go over this better:
https://www.librechat.ai/docs/configuration/authentication#user-management-scripts
